packages/icons/tasks/scaffold-metadata.js

```diff
@@ -70,14 +78,54 @@ async function scaffold() {
     {}
   );
 
+  /**
+   * build object that maps icon names to category & subcategory
+   *   - loop through all of `categories.yml`
+   *   - keys are icon names
+   *   - each iconName object has props for category & subcategory
+   *
+   * when needing to find category information,
+   * `categoryInformation[iconName].category` and
+   * `categoryInformation[iconName].subcategory` will return needed info
+   */
+  let categoryInformation = {};
+  categoriesJson['categories'].forEach(category => {
+    category.subcategories.forEach(subcategory => {
+      subcategory.members.forEach(iconName => {
+        categoryInformation[iconName] = {
+          category: category.name,
+          subcategory: subcategory.name,
+        };
+      });
+    });
+  });
+
   const icons = Object.keys(iconsGroupedByName).map(key => {
+    let iconIsCategorized =
+      categoryInformation[key] &&
+      categoryInformation[key].hasOwnProperty('subcategory');
```

Comment:
Do we need the `hasOwnProperty` check here? Or will this work:

```suggestion
      categoryInformation[key].subcategory
```